src/core/processing/qgsprocessingutils.h

```diff
@@ -224,6 +224,14 @@ class CORE_EXPORT QgsProcessingUtils
         QgsProcessingContext &context,
         QgsProcessingFeedback *feedback );
 
+    /**
+     * Combines two field lists, avoiding duplicate field names (in a case-insensitive manner).
+     *
+     * Duplicate field names will be altered to "name_2", "name_3", etc, finding the first
+     * non-duplicate name.
```

Comment:
Maybe it is worth to add that some formats (e.g. shapefiles) have restrictions for field names and resutling names may be truncated?